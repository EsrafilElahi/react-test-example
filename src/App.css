.App {
  text-align: center;
}

.hr {
  margin: 30px 0;
}

button {
  padding: 10px;
  border: 1px solid rgb(170, 172, 52);
  outline: none;
  background: rgb(105, 221, 211);
  margin-right: 5px;
  border-radius: 2px;
  cursor: pointer;
}
