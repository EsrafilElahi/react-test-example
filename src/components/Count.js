import React, { useState } from 'react'


function Count() {

    const [count, setCount] = useState(0)

    const increment = () => {
        setCount(prevState => prevState + 1)
    }

    const decrement = () => {
        setCount(prevState => prevState - 1)
    }

    return (
        <div>
            <h2>count: {count}</h2>
            <button className='inc' onClick={increment}>+</button>
            <button className='dec' onClick={decrement}>-</button>

            <hr className='hr' />
        </div>
    )
}

export default Count
