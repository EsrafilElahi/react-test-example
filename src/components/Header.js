import React, { useState } from 'react'


function Header() {

    const [h1, setH1] = useState('h1')

    const handleAlert = () => {
        setH1('hello')
    }

    return (
        <div>
            <h2 className='title'>Hello World</h2>
            <p className='paragraph'>react unit testing</p>
            <p className='h1'>hey:{h1}</p>
            <button className='alert' onClick={handleAlert}>submit</button>

            <hr className='hr' />
        </div>
    )
}

export default Header
