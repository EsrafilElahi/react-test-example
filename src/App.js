import { useState } from 'react'
import './App.css';
import Header from './components/Header';
import Count from './components/Count';



function App() {

  return (
    <div className="App">

      <Header />

      <Count />

    </div>
  );
}

export default App;
