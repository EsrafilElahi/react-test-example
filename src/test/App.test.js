import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from '../App'
import Header from './../components/Header';
import Count from './../components/Count';



// ==========================================
describe('>> App render', () => {

    let headerWrapper;
    let countWrapper;
    beforeEach(() => {
        headerWrapper = shallow(<Header />);
        countWrapper = shallow(<Count />);
    });

    it('title h2 ---> hello world', () => {
        const h2 = headerWrapper.find('.title')
        expect(h2.text()).toBe('Hello World')
    })
    it('paragraph ---> to Equal', () => {
        const p = headerWrapper.find('.paragraph')
        expect(p.text()).toEqual('react unit testing')
    })
    it('search btns ---> btn alert', () => {
        const btnAlert = headerWrapper.find('button')
        expect(btnAlert.at(0).props().className).toBe('alert')
    })
    it('hr ---> have class', () => {
        const hr = headerWrapper.find('hr')
        expect(hr.exists('.hr')).toBe(true)
    })
    it('props in ---> inc btn', () => {
        const btnAlert = countWrapper.find('.inc')
        expect(btnAlert.props().className).toEqual('inc')
    })
    it('btn alert ---> text', () => {
        expect(headerWrapper.find('.alert').text()).toBe("submit")
    })

})

// ==========================================
describe('>> simulate click', () => {

    let headerWrapper;
    let countWrapper;
    beforeEach(() => {
        headerWrapper = shallow(<Header />)
        countWrapper = shallow(<Count />)
    })

    test('click ---> submit', () => {
        const btnAlert = headerWrapper.find('button')
        btnAlert.simulate('click')
        expect(headerWrapper.find('.h1').text()).toBe('hey:hello')
    });
    test('h2 ---> count', () => {
        const h2 = countWrapper.find('h2')
        expect(h2.at(0).text()).toEqual('count: 0')
    })
    test('click ---> increment', () => {
        const btnInc = countWrapper.find('.inc')
        btnInc.simulate('click')
        const h2 = countWrapper.find('h2')
        expect(h2.at(0).text()).toEqual('count: 1')
    })
    test('click ---> decrement', () => {
        const btnDec = countWrapper.find('button').at(1)
        btnDec.simulate('click')
        const h2 = countWrapper.find('h2')
        expect(h2.at(0).text()).toEqual('count: -1')

    })
})

// ==========================================
describe('>> snapShots', () => {

    test('Header ---> snapShots', () => {
        const wrapper = shallow(<Header />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('Count ---> snapShots', () => {
        const wrapper = shallow(<Count />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})
