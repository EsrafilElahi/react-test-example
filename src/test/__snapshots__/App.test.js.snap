// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> snapShots App snapShots 1`] = `
<div
  className="App"
>
  <h2
    className="title"
  >
    Hello World
  </h2>
  <p
    className="paragraph"
  >
    react unit testing
  </p>
  <p
    className="h1"
  >
    h1: 
    h1
  </p>
  <button
    className="alert"
    onClick={[Function]}
  >
    submit
  </button>
  <hr
    className="hr"
  />
  <h2>
    count: 
    0
  </h2>
  <button
    className="inc"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="dec"
    onClick={[Function]}
  >
    -
  </button>
  <hr
    className="hr"
  />
</div>
`;

exports[`>> snapShots App snapShots 2`] = `
<div>
  <h2
    className="title"
  >
    Hello World
  </h2>
  <p
    className="paragraph"
  >
    react unit testing
  </p>
  <p
    className="h1"
  >
    hey:
    h1
  </p>
  <button
    className="alert"
    onClick={[Function]}
  >
    submit
  </button>
  <hr
    className="hr"
  />
</div>
`;

exports[`>> snapShots Count snapShots 1`] = `
<div>
  <h2>
    count: 
    0
  </h2>
  <button
    className="inc"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="dec"
    onClick={[Function]}
  >
    -
  </button>
  <hr
    className="hr"
  />
</div>
`;

exports[`>> snapShots Header ---> snapShots 1`] = `
<div>
  <h2
    className="title"
  >
    Hello World
  </h2>
  <p
    className="paragraph"
  >
    react unit testing
  </p>
  <p
    className="h1"
  >
    hey:
    h1
  </p>
  <button
    className="alert"
    onClick={[Function]}
  >
    submit
  </button>
  <hr
    className="hr"
  />
</div>
`;

exports[`snapShots App snapShots 1`] = `
<div
  className="App"
>
  <h2
    className="title"
  >
    Hello World
  </h2>
  <p
    className="paragraph"
  >
    react unit testing
  </p>
  <button
    className="alert"
    onClick={[Function]}
  >
    submit
  </button>
  <hr
    className="hr"
  />
  <h2>
    count : 
    0
     
  </h2>
  <button
    className="inc"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="dec"
    onClick={[Function]}
  >
    -
  </button>
  <hr
    className="hr"
  />
</div>
`;
